src/api/include/migraphx/migraphx.hpp

```diff
@@ -310,13 +346,21 @@ struct interface_base : Base
 
     protected:
     template <class F>
-    static migraphx_status try_(F f) // NOLINT
+    static migraphx_status try_(F f, const char** ex_msg = nullptr) // NOLINT
     {
         try
         {
             f();
             return migraphx_status_success;
         }
+        catch(std::exception& ex)
+        {
+            if(ex_msg)
+            {
+                *ex_msg = std::move(ex.what());
```

Comment:
This is still a dangling reference. Upon returning, `ex` lifetime ends which means any pointers to `ex` or its internal members(which is what `ex.what()` is returning) is no longer valid.

You need to copy the string to the buffer: `std::strncpy(&ex_msg, ex.what(), ex_msg_size)`

Reply:
Done